### PR TITLE
Use hatchling over setuptools and tox to orchestrate Python env setup

### PR DESCRIPTION
**Pull request by gaborbernat**
https://tox.wiki/en/latest/ is a project that automates Python environment setup for multiple Python versions and the de facto standard in Python world. hatchling is a modern replacement for setuptools.

### CODE REVIEW

File: client/python/README.md [hunk 1]
```diff
@@ -3,22 +3,22 @@
 To install from source, run:
 
 ```bash
-$ python setup.py install
```
Comment (gaborbernat) [inline client/python/README.md:6]: As per https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html this is not supported anymore.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      - id: check-merge-conflict
+      - id: check-toml
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: "v0.0.265"
+    hooks:
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
```
Comment (JDarDagran) [inline .pre-commit-config.yaml:17]: How about we add `insert-license` from `repo: https://github.com/Lucas-C/pre-commit-hooks`? We can do that in separate PR as well, just seems a nice addition while introducing pre-commit at all.

**Review by JDarDagran — APPROVED** [review]
@gaborbernat, that's a great PR. Thank you for your contribution. If you could just fix one missing header from the check.

@mobuchowski we should also follow same for `openlineage-airflow`

File: .pre-commit-config.yaml (hunk 2 above)
Comment (gaborbernat) [reply .pre-commit-config.yaml:17]: Would do it as follow-up as is orthogonal  👍

**Comment by gaborbernat** [comment]
@JDarDagran done 👍

**Review by mobuchowski — APPROVED** [review]

File: integration/airflow/tests/integration/Dockerfile [hunk 3]
```diff
@@ -20,7 +20,7 @@ RUN AIRFLOW_VERSION=`airflow version` && \
     cd ../airflow && \
     pip install --no-cache-dir --use-deprecated=legacy-resolver -r dev-requirements.txt \
     --constraint=https://raw.githubusercontent.com/apache/airflow/constraints-$AIRFLOW_VERSION/constraints-$PYTHON_VERSION.txt
-RUN python -m virtualenv dbt_venv && pwd && source dbt_venv/bin/activate &&  \
+RUN python -m virtualenv dbt_venv && pwd && source dbt_venv/bin/activate && pip install -U pip && \
     unset PIP_USER && \
     cd /app/openlineage/integration/sql && \
     pip install openlineage-sql --no-index --find-links target/wheels --force-reinstall && \
```
Comment (JDarDagran) [inline integration/airflow/tests/integration/Dockerfile:26]:
```suggestion
RUN python -m virtualenv dbt_venv && pwd && source dbt_venv/bin/activate && \
    unset PIP_USER && \
    python -m pip install pip --upgrade && \
    cd /app/openlineage/integration/sql && \
    pip install openlineage-sql --no-index --find-links target/wheels --force-reinstall && \
```

Seems to do the job for Airflow 2.1.4 integration test build

**Review by JDarDagran — APPROVED** [review]
@gaborbernat, thanks again for the PR! LGTM 👍
